src/actions/languages/python/motion.ts

```diff
@@ -0,0 +1,256 @@
+   * Filter function that returns true if an element is strictly ahead (false if equal)
+   * of the specified (first arg) position.
+   */
+  static isAhead(position: MinimalPosition, elementPosition: MinimalPosition) {
+    return (
+      elementPosition.line > position.line ||
+      (elementPosition.line === position.line && elementPosition.character > position.character)
+    );
+  }
+
+  /*
+   * Filter function that returns true if an element is strictly behind (false if equal)
+   * of the specified (first arg) position.
+   */
+  static isBehind(position: MinimalPosition, elementPosition: MinimalPosition) {
+    return (
+      elementPosition.line < position.line ||
+      (elementPosition.line === position.line && elementPosition.character < position.character)
+    );
+  }
+
+  /*
+   * Find the position of the specified:
+   *    type: function or class
+   *    direction: next or prev
+   *    edge: start or end
+   *
+   * With this information one can determine all of the required motions
+   */
+  find(type: Type, direction: Direction, edge: Edge, position: Position): Position | null {
+    // Choose the filtering utility based on direction
+    const isDirection = direction === 'next' ? PythonDocument.isAhead : PythonDocument.isBehind;
+
+    // Filter function for all elements whose "edge" is in the correct "direction"
+    // relative to the cursor's position
+    const dir = (element: StructureElement) => isDirection(position, element[edge]);
+
+    // Filter out elements from structure based on type and direction
+    const elements = this.structure.filter((elem) => elem.type === type).filter(dir);
+
+    if (edge === 'end') {
+      // When moving to an 'end' the elements should be started by the end position
+      elements.sort((a, b) => a.end.line - b.end.line);
+    }
+
+    // Return the first match if any exist
+    if (elements.length) {
+      // If direction === 'next' return the first element
+      // otherwise return the last element
+      const index = direction === 'next' ? 0 : elements.length - 1;
+      const element = elements[index];
+      const pos = element[edge];
+
+      // execAction MUST return a fully realized Position object created using new
+      return new Position(pos.line, pos.character);
+    }
+
+    return null;
+  }
+
+  // Use PythonDocument instance to move to specified class boundary
+  static moveClassBoundary(
+    document: TextDocument,
+    position: Position,
+    forward: boolean,
+    start: boolean
+  ): Position {
+    const direction = forward ? 'next' : 'prev';
+    const edge = start ? 'start' : 'end';
+
+    return new PythonDocument(document).find('class', direction, edge, position) || position;
+  }
+}
+
+type Type = 'function' | 'class';
+type Edge = 'start' | 'end';
+type Direction = 'next' | 'prev';
+
+// Uses the specified findFunction to execute the motion coupled to the shortcut (keys)
+abstract class BasePythonMovement extends BaseMovement {
+  type: Type;
```

Comment:
Make these properties abstract

Reply:
This should fix the build error - I recently enabled `strictPropertyInitialization`.

Reply:
Done.